websockify/websocket.py

```diff
@@ -310,17 +323,8 @@ def send_frames(self, bufs=None):
 
         if bufs:
             for buf in bufs:
-                if self.base64:
-                    encbuf, lenhead, lentail = self.encode_hybi(buf, opcode=1, base64=True)
-                else:
-                    encbuf, lenhead, lentail = self.encode_hybi(buf, opcode=2, base64=False)
-
-                if self.rec:
-                    self.rec.write("%s,\n" %
-                            repr("{%s{" % tdelta
-                                + encbuf[lenhead:len(encbuf)-lentail]))
-
-                self.send_parts.append(encbuf)
+                opcode = 2-int(self.base64)     #based64: opcode=1, binary: opcode=2
+                encbufs = self.send_hybi(buf, opcode, base64=self.base64, record=self.rec)
 
```

Comment:
You dropped the recording functionality. Also, the condition opcode calculation, while more concise, obfuscates the actual logic that is happening; the original is preferable to me.